### DISCUSSION TITLE
How to send live audio Inputs via discord bots?

### DISCUSSION THREAD

**Discussion by Carlimero12** [Rapptz/discord.py#10036] [Q&A]
I am working on a discord bot client which can set text via tk interface BUT i am stuck now. I dont know ho i transfer live audio input from headsets or micros to voice channels live. Please If someone can help me.

**Comment by exorcize** [comment]
Can you elaborate?

You can't live transfer the audio transmitted directly into the headset, the library won't have access to these inputs, you'll probably need to set up a pipeline to process this audio, recording it with `ffmpeg` or some such utility and then use it in your bot

**Comment by ghost** [comment]
It probably depends on the audio source. As I said; i got it working with connecting my phone with auxiliary to the auxiliary input (on my desktop computer) and then using a class formatted to ffmpeg to replay the audio on the discord bot. What do you use as an audio source?(I run the discord bot locally)On 11 Jul 2025, at 5:14 PM, 鏡花水月 ***@***.***> wrote:﻿
Can you elaborate?
You can't live transfer the audio transmitted directly into the headset, the library won't have access to these inputs, you'll probably need to set up a pipeline to process this audio, recording it with ffmpeg or some such utility and then use it in your bot

**Reply by Carlimero12** [reply]
Csn you record inputs like start record and stop then save as file and send via dc?

**Reply by Carlimero12** [reply]
Thanks
